proposals/lip-Introduce_cross_chain_messages.md

```diff
@@ -0,0 +1,650 @@
+We are following an optimistic approach in which all messages will be delivered and error handling will kick in when it is needed. 
+In that regard, users should be aware that valid messages will usually not trigger a response. 
+A quick API call will guarantee that the message has been properly received and executed.
+ 
+For a given module, the set of command IDs and cross-chain command IDs are allowed to overlap.
+
+
+#### Fee
+
+For all cross-chain messages, a fee paid in LSK is used to account for the transaction processing in the recipient chain.
+This fee must be transferred from the sending chain account to the recipient chain account in order to maintain the correct LSK balances on all chains in the ecosystem. 
+The [token ID][token-LIP-tokenID] for this fee is always the token ID of the LSK token and as such is not repeated in the message. 
+The LSK token is the main utility token of the Lisk ecosystem and as such is the only good candidate for paying cross-chain fees.
+
+
+#### Status
+
+The basic error handling for routing messages to other chains is done by the mainchain. 
+For example, in the case the recipient chain does not exist, is not active or has been terminated, the mainchain will return the message to the sending chain. 
+The sending chain can then revert the message and potentially refund users. 
+This design choice allows sidechains to send messages to other chains without needing to monitor the status (or even existence) of every other chain. 
+Information about the reason why the message failed and the initial message identifier are stored in the `status` property.
+
+The constant table lists the different status codes defined by the Interoperability module. 
+To allow for future updates or improvements of the Interoperability module, we restrict other modules from using status codes up to 63 (included). 
+Other modules may use other status codes larger or equal to 64.
+
+
+#### Parameters
+
+The `params` property of the messages is defined by each module and can follow any schema, similar to the `params` property of a transaction. 
+The `params` property is not deserialized or validated by the Interoperability module.
+
+In the Lisk ecosystem, all cross-chain messages are routed through the mainchain. 
+This means that messages should always have a sufficiently small size in order to be easily included in mainchain blocks. 
+As the mainchain payload size limit is 15 KiB, and [other properties][CCU-LIP] in the cross-chain updates will not be larger than 4 KiB, we limit the message size to 10 KiB. 
+To guarantee that all messages can be included and handled, sidechains in the Lisk ecosystem should have a payload size limit equal to or greater than 15 KiB (15 x 1024 bytes).
+
+
+### Message ID
+
+As all messages emitted in the Lisk ecoystem are distinct (thanks to the `index` property), we can assign a unique ID to them.
```

Comment:
Don't you also need the sending chain to completely identify it? Maybe we could add that: "All messages in the Lisk ecosystem are uniquely identified by their `index` and `sendingChain` properties." or something similar

Reply:
modified the sentence and added something similar to your suggestion to the `Index` rationale.